cmd/server/cadence/server.go

```diff
@@ -220,8 +221,13 @@ func (s *server) startService() common.Daemon {
 		}
 	}
 
-	dispatcher, err := params.DispatcherProvider.Get(common.FrontendServiceName, s.cfg.PublicClient.HostPort)
+	// will return empty array if not enabled
+	privateKey, err := s.cfg.Authorization.OAuthAuthorizer.GetPrivateKey()
```

Comment:
like the comment, use clusterMetadata.clusterInfomation.<currentCluster>.AuthorizationProvider 

You may need to create a new wrapper to create `AuthProvider` based on the config as it will be used in `clientBean.go`